pumpkin/src/server/translation.rs

```diff
@@ -0,0 +1,201 @@
+#![expect(dead_code)]
+
+use std::{
+    fs::File,
+    io::{BufRead, BufReader},
+    path::PathBuf,
+};
+
+use pumpkin_config::TranslationConfig;
+use pumpkin_core::text::{style::Style, TextComponent, TextContent};
+use serde_json::Value;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TranslationError {
+    #[error("File does not exist")]
+    NoFileFound,
+    #[error("File cannot be opened.")]
+    InvalidFile,
+    #[error("Failed to read file. Error: {0}")]
+    FileRead(std::io::Error),
+    #[error("Invalid JSON encountered. Use only objects in the translation file.")]
+    JsonParse,
+}
+
+pub fn translate<'a>(
+    config: &TranslationConfig,
+    message: &'a str,
+) -> Result<TextComponent<'a>, TranslationError> {
+    if !config.enabled {
+        let path = "assets/lang/en_us/en_us.json";
+        let translations = get_translations(path, message)?;
```

Comment:
I think this should be cached, You can use LazyStatic

Reply:
I am not sure if you're talking about the crate or static with LazyLock, but I'd assume it would be the latter?